src/js/libgeo/GeoOps.js

```diff
@@ -371,6 +441,31 @@ geoOps.Free.updatePosition = function(el) {
     putStateComplexVector(param); // copy param
     el.homog = General.withUsage(param, "Point");
 };
+geoOps.Free.getRandomMove = function(el) {
+    var rcomplex = function() {
+        var fact = 0.1;
+        return CSNumber.complex((Math.random() - 0.5) * fact, (Math.random() - 0.5) * fact);
+    };
+    var oldpos = List.normalizeMax(el.homog);
+    var oz = oldpos.value[2];
+    var ozabs = CSNumber.abs(oz).value.real;
```

Comment:
I'm not sure whether we should prefer `abs2` over `abs` in order to avoid the square root. From a performance standpoint this would make sense, since it's easy enough to adjust the threshold accordingly. But from a maintainance standpoint the version with `abs` may be easier to understand. And the cost of `sqrt` likely isn't excessive, compared to all the other things we do.